modules/chromeAiRtdProvider.js

```diff
@@ -0,0 +1,421 @@
+import { submodule } from '../src/hook.js';
+import { logError, mergeDeep, logMessage, deepSetValue, deepAccess } from '../src/utils.js';
+import { getCoreStorageManager } from '../src/storageManager.js';
+
+/* global LanguageDetector, Summarizer */
+/**
+ * @typedef {import('../modules/rtdModule/index.js').RtdSubmodule} RtdSubmodule
+ */
+
+export const CONSTANTS = Object.freeze({
+  SUBMODULE_NAME: 'chromeAi',
+  REAL_TIME_MODULE: 'realTimeData',
+  LOG_PRE_FIX: 'ChromeAI-Rtd-Provider:',
+  STORAGE_KEY: 'chromeAi_detected_data', // Single key for both language and keywords
+  MIN_TEXT_LENGTH: 20,
+  DEFAULT_CONFIG: {
+    languageDetector: {
+      enabled: true,
+      confidence: 0.8,
+      ortb2Path: 'site.content.language' // Default path for language
+    },
+    summarizer: {
+      enabled: false,
+      type: 'headline', // 'headline' or 'paragraph'
+      format: 'markdown', // 'markdown' or 'plaintext'
+      length: 'short', // 'short', 'medium', or 'long'
+      ortb2Path: 'site.content.keywords', // Default path for keywords
+      cacheInLocalStorage: true // Whether to cache detected keywords in localStorage
```

Comment:
wouldnt the summary change from pageview to pageview? I am not sure I follow the utility of the cache

Reply:
@patmmccann We're caching the summary based on the full URL, not just the domain. This means that different pages on a news site can have unique summaries. Since the API takes some time to generate the summary, caching the processed data ensures we don't miss the opportunity to include it in the bidRequest. By storing the result, we avoid redundant processing and improve overall efficiency.

Reply:
Makes sense ty!